app/shared/AddressAutocomplete.js

```diff
@@ -0,0 +1,59 @@
+'use client'
+import { useEffect, useState } from 'react'
+import TextField from '@shared/inputs/TextField'
+import { usePlacesWidget } from 'react-google-autocomplete'
+import { NEXT_PUBLIC_GOOGLE_MAPS_KEY } from 'appEnv'
+
+const MAPS_API_KEY = NEXT_PUBLIC_GOOGLE_MAPS_KEY
+
+export default function AddressAutocomplete({ value, onChange }) {
+  const [inputValue, setInputValue] = useState(value || '')
+
+  const { ref } = usePlacesWidget({
+    apiKey: MAPS_API_KEY,
+    onPlaceSelected: (place) => {
+      if (place && place.formatted_address) {
+        setInputValue(place.formatted_address)
+        onChange(place.formatted_address, place)
```

Comment:
why pass both of these as args when the former is just a property of the latter? Any consumer can just easily read the property off the object argument, no?